src/vm/vm_memory/memory.rs

```diff
@@ -71,9 +81,14 @@ impl Memory {
         let relocatable: Relocatable = key
             .try_into()
             .map_err(|_| MemoryError::AddressNotRelocatable)?;
-        let (i, j) = from_relocatable_to_indexes(relocatable);
-        if self.data.len() > i && self.data[i].len() > j {
-            if let Some(ref element) = self.data[i][j] {
+        let data = if relocatable.segment_index.is_negative() {
+            &self.temp_data
+        } else {
+            &self.data
+        };
+        let (i, j) = from_relocatable_to_indexes(&relocatable);
```

Comment:
Is `from_relocatable_to_indexes` fixing the negative index? It should be something like `-(segment_index+1)`.